utils/log.cpp

```diff
@@ -106,8 +113,94 @@ class UnixLogger : public Logger {
     }
 };
 
+#endif
+
+/* File logger. */
+
+class FileLogger : public Logger {
+public:
+
+    FileLogger() {
+        const char *file_path = getenv("JSRDBG_LOG_FILE_PATH");
+        if (file_path) {
+            _out = std::ofstream(file_path, std::ofstream::out | std::ios::binary);
+        }
+    };
+
+    virtual ~FileLogger() {
+        _out.close();
+    }
+
+    enum Level {
+        INFO,
+        DEBUG,
+        WARN,
+        ERROR
+    };
+
+    void debug( const std::string log, ... ) {
+        va_list args;
+        va_start( args, log );
+        logMessage( DEBUG, log, args );
+        va_end( args );
+    }
+
+    void info( const std::string log, ... ) {
+        va_list args;
+        va_start( args, log );
+        logMessage( INFO, log, args );
+        va_end( args );
+    }
+
+    void warn( const std::string log, ... ) {
+        va_list args;
+        va_start( args, log );
+        logMessage( WARN, log, args );
+        va_end( args );
+    }
+
+    void error( const std::string log, ... ) {
+        va_list args;
+        va_start( args, log );
+        logMessage( ERROR, log, args );
+        va_end( args );
+    }
+
+private:
+    std::ofstream _out;
```

Comment:
Shouldn't we synchronize access to this stream? Loggers are supposed to be thread-safe.